src/ucp/wireup/select.c

```diff
@@ -306,6 +306,28 @@ ucp_wireup_init_select_info(double score, unsigned addr_index,
     select_info->priority   = priority;
 }
 
+/**
+ * Get bitmap of memory types that Memory Domain can be registered with taking
+ * into account context's maps of Memory Domains that provide registration for
+ * given memory type.
+ */
+static uint64_t ucp_wireup_select_reg_mem_types(uct_md_attr_t *md_attr,
+                                                ucp_context_h context,
+                                                ucp_rsc_index_t rsc_index)
+{
+    uint64_t reg_mem_types  = md_attr->cap.reg_mem_types;
```

Comment:
do we really need md_attr here?
maybe start with 0 and add bits from reg_md_map[mem_type] (instead of removing them)?